src/core/monte_carlo_scan_matcher.h

```diff
@@ -10,15 +20,40 @@
 #include "grid_scan_matcher.h"
 
 // TODO: move publish transform to observer
+
+/**
+ * \brief Scan Matcher that uses Mone Carlo Method.
+ * The focus of scan matcher is to compare a scan and a built map; in this
+ * class the method of scan comparison is performed by comparing corresponding
+ * scan costs.
+ */
 class MonteCarloScanMatcher : public GridScanMatcher {
 public:
   using ObsPtr = std::shared_ptr<GridScanMatcherObserver>;
 public:
+
+  /**
+   * \brief Initializes the scan matcher with a certain scan cost estimator.
+   * \param estimator Current estimator of Scan Cost.
+   * \param failed_iter Maximum amount of comparison of costs for scan at each
```

Comment:
Consider: A limit of generated samples that have higher cost comparing with the best estimated pose.